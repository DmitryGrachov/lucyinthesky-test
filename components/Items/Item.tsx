import { IOneItem } from '../../interfaces/items.interface';

const Item = ({ name, image }: IOneItem) => (
    <>
        <div>
            <img src={image} alt="mainImage" />
            <p>{name}</p>
        </div>
    </>
);

export default Item;
