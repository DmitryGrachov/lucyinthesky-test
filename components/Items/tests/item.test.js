import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Item from '../Item';

describe('Item component', () => {
    let expectedProps;

    beforeEach(() => {
        expectedProps = {
            name: 'Iridescent Sequin Bodycon Dress in Pink',
            altImage: 'mainImage',
        };
    });

    test('should render name and image', () => {
        const { getByText, getByAltText } = render(<Item {...expectedProps} />);
        const name = getByText(expectedProps.name);
        const altImage = getByAltText(expectedProps.altImage);

        expect(name).toBeVisible();
        expect(altImage).toBeVisible();
    });

    test('matches item snapshot', () => {
        const { asFragment } = render(<Item {...expectedProps} />);

        expect(asFragment()).toMatchSnapshot();
    });
});
