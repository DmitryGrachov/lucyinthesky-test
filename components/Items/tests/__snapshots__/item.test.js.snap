// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component matches item snapshot 1`] = `
<DocumentFragment>
  <div>
    <img
      alt="mainImage"
    />
    <p>
      Iridescent Sequin Bodycon Dress in Pink
    </p>
  </div>
</DocumentFragment>
`;
